crates/holochain_types/src/app.rs

```diff
@@ -200,1001 +180,3 @@ impl InstallAppDnaPayload {
-            "app",
-            agent.clone(),
-            vec![(role_name.clone(), role1)],
-            manifest,
-        )
-        .unwrap()
-        .into();
-
-        // Can add clones up to the limit
-        let clones: Vec<_> = vec![new_clone(), new_clone(), new_clone()];
-        let clone_id_0 = app.add_clone(&role_name, &clones[0]).unwrap();
-        let clone_id_1 = app.add_clone(&role_name, &clones[1]).unwrap();
-        let clone_id_2 = app.add_clone(&role_name, &clones[2]).unwrap();
-
-        assert_eq!(clone_id_0, CloneId::new(&role_name, 0));
-        assert_eq!(clone_id_1, CloneId::new(&role_name, 1));
-        assert_eq!(clone_id_2, CloneId::new(&role_name, 2));
-
-        assert_eq!(
-            app.clone_cell_ids().collect::<HashSet<_>>(),
-            maplit::hashset! { &clones[0], &clones[1], &clones[2] }
-        );
-        assert_eq!(app.clone_cells().count(), 3);
-
-        // Adding the same clone twice should return an error
-        let result_add_clone_twice = app.add_clone(&role_name, &clones[0]);
-        assert!(result_add_clone_twice.is_err());
-
-        // Adding a clone beyond the clone_limit is an error
-        matches::assert_matches!(
-            app.add_clone(&role_name, &new_clone()),
-            Err(AppError::CloneLimitExceeded(3, _))
-        );
-
-        // Disable a clone cell
-        app.disable_clone_cell(&clone_id_0).unwrap();
-        // Assert it is moved to disabled clone cells
-        assert!(!app
-            .clone_cells()
-            .any(|(clone_id, _)| *clone_id == clone_id_0));
-        assert_eq!(app.clone_cells().count(), 2);
-        assert!(app
-            .disabled_clone_cells()
-            .any(|(clone_id, _)| *clone_id == clone_id_0));
-
-        // Enable a disabled clone cell
-        let enabled_cell = app.enable_clone_cell(&clone_id_0).unwrap();
-        assert_eq!(
-            enabled_cell.role_name,
-            clone_id_0.as_app_role_name().to_owned()
-        );
-
-        // Enabling an already enabled cell does nothing.
-        let enabled_cell_2 = app.enable_clone_cell(&clone_id_0).unwrap();
-        assert_eq!(enabled_cell_2, enabled_cell);
-
-        // Assert it is accessible from the app again
-        assert!(app
-            .clone_cells()
-            .find(|(clone_id, _)| **clone_id == clone_id_0)
-            .is_some());
-        assert_eq!(
-            app.clone_cell_ids().collect::<HashSet<_>>(),
-            maplit::hashset! { &clones[0], &clones[1], &clones[2] }
-        );
-        assert_eq!(app.clone_cells().count(), 3);
-
-        // Disable and delete a clone cell
-        app.disable_clone_cell(&clone_id_0).unwrap();
-        // Disabling is also idempotent
-        app.disable_clone_cell(&clone_id_0).unwrap();
-
-        app.delete_clone_cell(&clone_id_0).unwrap();
-        // Assert the deleted cell cannot be enabled
-        assert!(app.enable_clone_cell(&clone_id_0).is_err());
-    }
-}
```

Comment:
the majority of changes in this PR come from moving most of this file into `holochain-app/src/ported.rs`